packages/opencensus-propagation-stackdriver/test/test-v1.ts

```diff
@@ -108,12 +91,47 @@ describe('inject', () => {
 
   testData.forEach(({input, output}) => {
     it(`returns well-formatted trace context for ${inspect(input)}`, () => {
-      const setter = {
-        setHeader(name: string, value: string) {
-          assert.deepEqual(value, output);
-        }
-      };
-      inject(setter, input);
+      const header = serializeSpanContext(input);
+      assert.deepEqual(header, output);
     });
   });
 });
+
+describe('extract', () => {
+  it('should return prased header from getter.getHeader', () => {
+    const HEADER = 'cafef00d/123';
+    const getter = {
+      getHeader(name: string) {
+        return HEADER;
+      }
+    };
+    assert.deepEqual(extract(getter), parseContextFromHeader(HEADER));
+  });
+});
+
+describe('inject', () => {
+  it('should call setter.setHeader with serialized header', (done) => {
+    const spanContext = generate();
+    const setter = {
+      setHeader(name: string, value: string) {
+        assert.deepEqual(value, serializeSpanContext(spanContext));
+        done();
+      }
+    };
+    inject(setter, spanContext);
+  });
+});
+
+describe('generate', () => {
+  it('should generate a valid span context', () => {
+    const TIMES = 20;
```

Comment:
Should we be checking uniqueness as well?

Reply:
Done.